Library/Homebrew/test/spec_helper.rb

```diff
@@ -260,6 +261,7 @@
       @__stdin.close
 
       Formulary.clear_cache
+      Tap.each(&:clear_cache)
```

Comment:
It seems like without modifying `Tap.each` you could do this instead or something similar. I have not tested this at all though.

```suggestion
      Tap.each(&:clear_cache)
      CoreTap.clear_cache
      CoreCaskTap.clear_cache
```

Reply:
Yes, it will work, but it doesn't make sense for `CoreTap` and `CoreCaskTap` not to already be included in `Tap.each`. In fact, they may actually be included, depending on whether the tap directory exists in tests, which is even weirder.

Reply:
This should actually be done before the `ENV` is restored above since this depends on `HOMEBREW_NO_INSTALL_FROM_API`.